src/executables/util_sa.c

```diff
@@ -1716,10 +1716,29 @@ diagdb (UTIL_FUNCTION_ARG * arg)
   if (diag == DIAGDUMP_ALL || diag == DIAGDUMP_HEAP)
     {
       bool dump_records;
-      /* this dumps the contents of all heaps */
+      const char *class_name;
       dump_records = utility_get_option_bool_value (arg_map, DIAG_DUMP_RECORDS_S);
-      fprintf (outfp, "\n*** DUMP OF ALL HEAPS ***\n");
-      (void) file_tracker_dump_all_heap (thread_p, outfp, dump_records);
+      class_name = utility_get_option_string_value (arg_map, DIAG_CLASS_NAME_S, 0);
```

Comment:
Why we have to check `class_name` option with `DIAGDUMP_ALL` option?
Does "dump all" not mean "dump all heaps"?
I think that if you want to check this option at this line, you should check about "If DIAGDUMP_ALL is on, class_name must be always off".

Reply:
I moved position for getting `class_name` parameter before splitting dump modes, and check if `diag==DIAGDUMP_ALL` and `class_name` is not `NULL`